zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/BackgroundTaskManager.java

```diff
@@ -94,21 +117,19 @@ public static BackgroundTaskManager create(
             .name("exporter-" + exporterId + "-p" + partitionId + "-tasks-", 0)
             .uncaughtExceptionHandler(FatalErrorHandler.uncaughtExceptionHandler(logger))
             .factory();
-    final var executor = defaultExecutor(threadFactory, partitionId);
+    final var executor = defaultExecutor(threadFactory);
     final var repository =
         createRepository(config, resourceProvider, partitionId, executor, metrics, logger);
-    final var archiver = new BackgroundTaskManager(partitionId, repository, logger, executor);
-    final var processInstanceJob =
-        createProcessInstanceJob(metrics, logger, resourceProvider, repository, executor);
+    final List<BackgroundTask> tasks = new ArrayList<>();
+
+    tasks.add(createProcessInstanceJob(metrics, logger, resourceProvider, repository, executor));
     if (partitionId == START_PARTITION_ID) {
-      final var batchOperationJob =
-          createBatchOperationJob(metrics, logger, resourceProvider, repository, executor);
-      archiver.start(config.getArchiver(), processInstanceJob, batchOperationJob);
-    } else {
-      archiver.start(config.getArchiver(), processInstanceJob);
+      tasks.add(createBatchOperationJob(metrics, logger, resourceProvider, repository, executor));
     }
 
-    return archiver;
+    executor.setCorePoolSize(Math.min(tasks.size(), MAX_BACKGROUND_THREADS));
```

Comment:
Nice work around to use a list of tasks 🙌

Reply:
I updated it to actually just set the max size, and keep core size to 0. Threads time out after a minute, meaning if they have nothing to do for a minute, they will shut down.

So this means when we only have one job, it should pretty much keep the thread alive all the time. When we have two jobs, if there isn't much to do, the jobs will be interleaved, and we should still only use one thread :+1: If we have more work and more jobs, then we'll have more threads.

Reply:
We could consider enforcing always having at minimum 1 thread anyway :shrug: